WikiWatershed/model-my-watershed
# WIP: Integrate sidebar

Implements the new sidebar design.

TODO:
- [X] Skip over analyze page
- [X] ~~Run models when tab is activated~~ There are some issues with this.
- [X] Merge analyze and modeling route preparation
- [X] Move analyze results to tab
- [x] Make analyze tabs style match modeling tabs
- [x] Fix modeling spinners (currently disappearing before model is finished)
- [x] Fix analyze messages
- [x] ~~Format charts and graphs according to wireframes~~ There are bunch of tasks that need to be done here. I've fixed what I could, and will create a list of the remaning issues for @jfrankl.
- [x] Start analyze when page is loaded
- [x] ~~Fix tests~~ None of them broke!

Update: Most of the work is done here. There is some UI polishing that needs to happen, but testing and review can begin.

**Testing instructions**
- Select an area of the interest.
- You should be forwarded directly to the modeling page, which is rendered using the new sidebar design.
- Interact with the app, and report and weird behavior or bugs created by moving to this design.

Known issues:

- Minor UI issues described in a comment below.
- The charts code probably use another pass. A lot of times, the ticks are overlapping, and their style doesn't quite match up with the wireframes (in some minor ways): https://marvelapp.com/ad6b6b#8306748
- Some model runs appear to be canceling each other out, but all the results appear to come back. It's not clear to me yet what's going on here. It happened more frequently when we only ran models when a modeling tab was activated. To avoid that, we now run them all at once like we did previously.

Connects to #1093.
